### PR TITLE
update jwt claim based routing and add tests

### PR DESCRIPTION
**Pull request by yangminzhu**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [x] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/tasks/security/authentication/jwt-route/index.md [hunk 1]
```diff
@@ -72,6 +72,12 @@ identity and more secure compared using the unauthenticated HTTP attributes (e.g
     The request authentication enables JWT validation on the Istio ingress gateway so that the validated JWT claims
     can later be used in the virtual service for routing purposes.
 
+    The request authentication is applied on the ingress gateway because the JWT claim based routing is only supported
+    on ingress gateways.
+
+    Note: the request authentication will only check the JWT if it exists in the request. To make the JWT required and
+    reject the request if it does not include JWT, apply the authorization policy as specifeid in the [task](/docs/tasks/security/authentication/authn-policy#require-a-valid-token).
```
Comment (ericvn) [inline content/en/docs/tasks/security/authentication/jwt-route/index.md:79]: s/specifeid/specified/

Comment (yangminzhu) [reply content/en/docs/tasks/security/authentication/jwt-route/index.md]: thanks.

**Comment by yangminzhu** [comment]
note, the test failed because it was using the old proxy, rerunning the test to see if it now uses the latest.

**Comment by ericvn** [comment]
Note that the current tests reference the 1.12.0-beta.0 build. I do have a PR open to move to the beta.1, just waiting on its approval.

**Comment by yangminzhu** [comment]
@ericvn tests are now passing, ptal, thanks.

**Review by ericvn — APPROVED** [review]

**Pull request merged**